enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+---
+title: single-node-deployment-with-bootstrap-in-place
+authors:
+  - "@eranco"
+  - "@mrunalp"
+  - "@dhellmann"
+  - "@romfreiman"
+  - "@tsorya"
+reviewers:
+  - TBD
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-13
+last-updated: 2020-12-13
+status: implementable
+see-also:
+  - https://github.com/openshift/enhancements/pull/560
+  - https://github.com/openshift/enhancements/pull/302
+---
+
+# Single Node deployment with bootstrap-in-place
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add support for new features such as [single-node production deployment](https://github.com/openshift/enhancements/pull/560/files),
+we need a way to install such clusters without an extra node dependency for bootstrap.
+
+This enhancement describes the flow for installing Single Node OpenShift using a liveCD that performs the bootstrap logic and reboots to become the single node.
+
+## Motivation
+
+Currently, all OpenShift installations use an auxiliary bootstrap node.
+The bootstrap node creates a temporary control plane that is required for launching the actual cluster.
+
+Single Node OpenShift installations will often be performed in environments where there are no extra nodes, so it is highly desirable to remove the need for a separate bootstrap machine to reduce the resources required to install the cluster.
+
+The downsides of requiring a bootstrap node for Single Node OpenShift are:
+
+1. The obvious additional node.
+2. Requires external dependencies:
+   a. Load balancer (only for bootstrap phase)
+   b. Requires DNS (configured per installation)
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node OpenShift
+
+### Goals
+
+* Describe an approach for installing Single Node OpenShift in a BareMetal environment for production use.
+* Minimal changes to OpenShift installer and the implementation shouldn't affect existing deployment flows.
+* Installation should result a clean Single Node OpenShift without any bootstrap leftovers.
+
+
+### Non-Goals
+
+* Addressing a similar installation flow for multi-node clusters.
+* High-availability for Single Node OpenShift.
+* Single-node-developer (CRC) cluster-profile installation.
+* Supporting cloud deployment for bootstrap in place (since livecd cannot be used). It will be addressed as part of future enhancement.
+* Upgrading Single Node OpenShift will be addressed as part of a future enhancement.
```

Comment:
I understand that this is a future goal but it has implications for implementation so thought I would start the converstaion. If the upgrade model is replace vs inplace upgrade does this mean that the etcd state is more or less ephemeral living only as long as the cluster version? Today we do no support backup/restore without at least one existing node from the previous cluster. It would be good to understand this generally as early as possible. cc @romfreiman

Reply:
@hexfusion not sure we know the answer right now. For 4.8, we want to explore of redeploying while preserving user data. Maybe use existing opensource tool to backup and restore - someone mentioned valero.
For in place upgrade there are 2 spikes for 4.8, and I assume going forward that would be the direction we will take. 
I think we should have a doc discussing upgrade options - there we can follow up on the backup/restore with pre-existing node as a requirement as well. @dhellmann WDYT?